content/en/docs/languages/go/instrumentation.md

```diff
@@ -164,23 +164,43 @@ Once a span has completed, it is immutable and can no longer be modified.
 
 ### Span Attributes
 
-Attributes are keys and values that are applied as metadata to your spans and
+[Attributes] are keys and values that are applied as metadata to your spans and
 are useful for aggregating, filtering, and grouping traces. Attributes can be
 added at span creation, or at any other time during the lifecycle of a span
 before it has completed.
 
+Generally speaking, you should try to set attributes during span creation. This
+will incur less overhead compared to setting attributes after span creation, and
```

Comment:
> This
> will incur less overhead compared to setting attributes after span creation

^ Is this actually true? I think the answer is - it depends.!

If the sampler is ParentBased and the parent's decision is to Drop, then providing all attributes at startup time is wasted as the span is going to be dropped anyway...
If user choses to add it after span creation, he/she can avoid the cost of representing attributes completes, by short-circuiting if span is not recording.

The only reason spec recommended providing attributes at creation time is for the trace-sampler to access it, not perf. If Golang implementation makes it more performant, then this section makes sense - but I wanted to double check if that is indeed true.

Reply:
This is the justification: https://cloud-native.slack.com/archives/CJFCJHG4Q/p1706816394957449 (SetAttributes also takes a variadic argument). My question would be if the Go compiler optimizes the single value case though.